packages/govern-core/contracts/pipelines/GovernQueue.sol

```diff
@@ -212,13 +212,32 @@ contract GovernQueue is IERC3000, AdaptativeERC165, IArbitrable, ACL {
         }
     }
 
-    function veto(bytes32 _containerHash, bytes memory _reason) auth(this.veto.selector) override public {
-        queue[_containerHash].checkAndSetState(
-            GovernQueueStateLib.State.Scheduled,
-            GovernQueueStateLib.State.Cancelled
-        );
+    function veto(
+        ERC3000Data.Container memory _container,
+        bytes memory _reason
+    ) auth(this.veto.selector) override public {
+        bytes32 containerHash = _container.hash();
+        GovernQueueStateLib.Item storage item = queue[containerHash];
+
+        if (item.state == GovernQueueStateLib.State.Challenged) {
+            item.checkAndSetState(
+                GovernQueueStateLib.State.Challenged,
+                GovernQueueStateLib.State.Cancelled
+            );
+
+            _container.config.challengeDeposit.releaseTo(challengerCache[containerHash]);
+            challengerCache[containerHash] = address(0);
+            delete disputeItemCache[containerHash][IArbitrator(_container.config.resolver)];
+        } else {
+            item.checkAndSetState(
```

Comment:
Also, here, it might be good leaving a comment on why do we only explicitly check for only the `Scheduled` state (as all other states are invalid and should revert).